packages/starlight/utils/createPathFormatter.ts

```diff
@@ -0,0 +1,58 @@
+import type { AstroConfig } from 'astro';
+import { fileWithBase, pathWithBase } from './base';
+import {
+	ensureHtmlExtension,
+	ensureTrailingSlash,
+	stripHtmlExtension,
+	stripTrailingSlash,
+} from './path';
+
+interface FormatPathOptions {
+	format?: AstroConfig['build']['format'];
+	trailingSlash?: AstroConfig['trailingSlash'];
+}
```

Comment:
Ah, maybe. `base` is it a bit different because you access it with `import.meta.env.BASE_URL` instead of via our project context virtual module. I think we can leave it for now.